cincoctrl/cincoctrl/findingaids/views.py

```diff
@@ -83,10 +82,28 @@ def get_form_kwargs(self, *args, **kwargs):
         kwargs["queryset"] = self.request.user.repositories()
         return kwargs
 
+    def extract_ead_fields(self, uploaded_file):
+        file_content = b""
+        for chunk in uploaded_file.chunks():
+            file_content += chunk
+        # reset file pointer for file save
+        uploaded_file.seek(0)
+        p = EADParser()
+        p.parse_string(file_content)
+        return p.extract_ead_fields()
+
     def form_valid(self, form):
         form.instance.created_by = self.request.user
         form.instance.record_type = "ead"
-        return super().form_valid(form)
+        if "ead_file" in self.request.FILES:
+            form.instance.collection_title, form.instance.collection_number = (
+                self.extract_ead_fields(self.request.FILES["ead_file"])
+            )
+        is_valid = super().form_valid(form)
+        if is_valid:
+            self.object.queue_index()
```

Comment:
What happens if the save fails for some reason, like the DB went away? Would index value be different to the db value? Or or does the queue_index query the DB itself?

Reply:
this logic is going away, I reorganized.